encoders/firebase-encoders-reflective/src/main/java/com/google/firebase/encoders/reflective/ReflectiveDecoderFieldContextImp.java

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.encoders.reflective;
+
+import com.google.firebase.decoders.FieldRef;
+import com.google.firebase.encoders.EncodingException;
+import com.google.firebase.encoders.FieldDescriptor;
+import com.google.firebase.encoders.annotations.Encodable;
+import com.google.firebase.encoders.annotations.ExtraProperty;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+class ReflectiveDecoderFieldContextImp<T> implements ReflectiveDecoderFieldContext<T> {
```

Comment:
nit: it's common to name these things with `Impl`, not `Imp`
```suggestion
class ReflectiveDecoderFieldContextImpl<T> implements ReflectiveDecoderFieldContext<T> {
```